src/Fixer/ClassNotation/StaticPrivateMethodFixer.php

```diff
@@ -0,0 +1,308 @@
+        }
+
+        $classClose = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $classOpen);
+        foreach ($this->getClassMethods($tokens, $classOpen, $classClose) as $methodData) {
+            list(, $methodOpen, $methodClose) = $methodData;
+
+            $this->fixReferencesInFunction($tokens, $tokensAnalyzer, $methodOpen, $methodClose, $fixedMethods);
+        }
+    }
+
+    /**
+     * @param Tokens         $tokens
+     * @param TokensAnalyzer $tokensAnalyzer
+     * @param int            $functionKeywordIndex
+     * @param int            $methodOpen
+     * @param int            $methodClose
+     *
+     * @return bool
+     */
+    private function skipMethod(Tokens $tokens, TokensAnalyzer $tokensAnalyzer, $functionKeywordIndex, $methodOpen, $methodClose)
+    {
+        $methodNameIndex = $tokens->getNextMeaningfulToken($functionKeywordIndex);
+        $methodName = strtolower($tokens[$methodNameIndex]->getContent());
+        if (isset($this->magicNames[$methodName])) {
+            return true;
+        }
+
+        $prevTokenIndex = $tokens->getPrevMeaningfulToken($functionKeywordIndex);
+        if (!$tokens[$prevTokenIndex]->isGivenKind(T_PRIVATE)) {
+            return true;
+        }
+
+        $prePrevTokenIndex = $tokens->getPrevMeaningfulToken($prevTokenIndex);
+        if ($tokens[$prePrevTokenIndex]->isGivenKind(T_STATIC)) {
+            return true;
+        }
+
+        for ($index = $methodOpen + 1; $index < $methodClose - 1; ++$index) {
+            if ($tokens[$index]->isGivenKind(T_CLASS) && $tokensAnalyzer->isAnonymousClass($index)) {
+                $anonymousClassOpen = $tokens->getNextTokenOfKind($index, ['{']);
+                $index = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $anonymousClassOpen);
+
+                continue;
+            }
+
+            if ($tokens[$index]->isGivenKind(T_FUNCTION)) {
+                return true;
+            }
+
+            if ($tokens[$index]->equals([T_VARIABLE, '$this'])) {
+                $operatorIndex = $tokens->getNextMeaningfulToken($index);
+                $methodNameIndex = $tokens->getNextMeaningfulToken($operatorIndex);
+                $argumentsBraceIndex = $tokens->getNextMeaningfulToken($methodNameIndex);
+
+                if (
+                    !$tokens[$operatorIndex]->isGivenKind(T_OBJECT_OPERATOR)
+                    || $methodName !== $tokens[$methodNameIndex]->getContent()
+                    || $tokens[$argumentsBraceIndex]->equals(['('])
+                ) {
+                    return true;
+                }
+            }
+
+            if ($tokens[$index]->equals([T_STRING, 'debug_backtrace'])) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param Tokens         $tokens
+     * @param TokensAnalyzer $tokensAnalyzer
+     * @param int            $methodOpen
+     * @param int            $methodClose
+     * @param array          $fixedMethods
```

Comment:
Since this array has to be of a specific key/value type, how about making that explicit in the types here:
```suggestion
     * @param array<string, bool> $fixedMethods
```

Reply:
Didn't know this library uses generic annotations, thank you for the feedback